pkg/operator/ceph/nfs/nfs.go

```diff
@@ -305,7 +306,17 @@ func (r *ReconcileCephNFS) configureNFSPool(n *cephv1.CephNFS) error {
 
 	_, err := cephclient.GetPoolDetails(r.context, r.clusterInfo, poolName)
 	if err != nil {
-		return errors.Wrapf(err, "pool %q could not be retrieved, ensure the pool is defined with a CephBlockPool", poolName)
+		// create default `.nfs` pool for versions >= ceph v16.2.7, if it does not exist.
+		if r.clusterInfo.CephVersion.IsAtLeast(cephNFSChangeVersion) && strings.Contains(err.Error(), "ENOENT") {
+			args := []string{"osd", "pool", "create", poolName}
+			output, err := cephclient.NewCephCommand(r.context, r.clusterInfo, args).Run()
```

Comment:
Let's finalize the discussion [here](https://github.com/rook/rook/pull/10014/files#r847746103) before considering merging this PR. My current perspective is that we should just require a .nfs CephBlockPool CR and not auto-create the pool.